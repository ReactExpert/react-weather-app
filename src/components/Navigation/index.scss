.navigation {
  align-items: center;
  display: flex;
  height: 20px;
  justify-content: center;
  margin-bottom: 20px;
  margin-top: 10px;
  width: 100%;

  &__dot {
    background-color: transparent;
    border: 1px solid #fff;
    border-radius: 6px;
    height: 6px;
    margin-left: 4px;
    margin-right: 4px;
    opacity: 0;
    transform: translateY(-20px);
    transition: transform .3s cubic-bezier(.165, .84, .44, 1), opacity .3s ease-out, background-color .3s ease-out;
    width: 6px;
  }

  &__dot.animate-in {
    opacity: 1;
    transform: translateY(0);
  }

  &.hourly {
    .navigation__dot:first-child {
      background-color: #fff;
    }

    .navigation__dot:last-child {
      background-color: transparent;
    }
  }

  &.daily {
    .navigation__dot:last-child {
      background-color: #fff;
    }

    .navigation__dot:first-child {
      background-color: transparent;
    }
  }
}
