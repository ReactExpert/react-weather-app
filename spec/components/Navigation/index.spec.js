import React from 'react';
import Navigation from '../../../src/components/Navigation';
import { enzymeConfig, mount } from '../../enzymeConfig';

enzymeConfig();

describe('Navigation', () => {
  it('matches snapshot with currentForecast hourly', () => {
    const component = mount(<Navigation currentForecast='hourly'/>);

    expect(component).toMatchSnapshot();
  });

  it('matches snapshot with currentForecast daily', () => {
    const component = mount(<Navigation currentForecast='daily'/>);

    expect(component).toMatchSnapshot();
  });

  describe('componentDidMount', () => {
    it('matches count of rays', () => {
      const component = mount(<Navigation currentForecast='hourly'/>);
      const instance = component.instance();

      expect(instance.dots.length).toEqual(2);
    });

    it('calls animate', (done) => {
      const component = mount(<Navigation currentForecast='hourly'/>);
      const instance = component.instance();

      instance.animate = jest.fn();

      setTimeout(() => {
        expect(instance.animate).toBeCalled();
        done();
      }, 400);
    });
  });

  describe('animate', () => {
    it('adds classes animate-in to dots', (done) => {
      const component = mount(<Navigation currentForecast='hourly' />);
      const instance = component.instance();

      instance.animate();

      setTimeout(() => {
        instance.dots.map( element => expect(element.classList).toContain('animate-in'));
        done();
      }, 1000);
    });
  });
});
