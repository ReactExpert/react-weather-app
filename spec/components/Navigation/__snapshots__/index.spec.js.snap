// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navigation matches snapshot with currentForecast daily 1`] = `
<Navigation
  currentForecast="daily"
>
  <section
    className="navigation daily"
  >
    <div
      className="navigation__dot"
    />
    <div
      className="navigation__dot"
    />
  </section>
</Navigation>
`;

exports[`Navigation matches snapshot with currentForecast hourly 1`] = `
<Navigation
  currentForecast="hourly"
>
  <section
    className="navigation hourly"
  >
    <div
      className="navigation__dot"
    />
    <div
      className="navigation__dot"
    />
  </section>
</Navigation>
`;
